Public/Src/Engine/UnitTests/Processes/SandboxedProcessBreakawayTest.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using BuildXL.Utilities;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using BuildXL.Processes;
+using Test.BuildXL.Executables.TestProcess;
+using Test.BuildXL.TestUtilities.Xunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Test.BuildXL.Processes
+{
+    public sealed class SandboxedProcessBreakawayTest : SandboxedProcessTestBase
+    {
+        public SandboxedProcessBreakawayTest(ITestOutputHelper output)
+            : base(output) { }
+
+        [TheoryIfSupported(requiresWindowsBasedOperatingSystem: true)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ChildProcessCanBreakawayWhenConfigured(bool letInfiniteWaiterSurvive)
+        {
+            // We use InfiniteWaiter (a process that waits forever) as a long-living process that we can actually check it can
+            // escape the job object
+            var fam = new FileAccessManifest(
+                Context.PathTable, 
+                childProcessesToBreakawayFromSandbox: letInfiniteWaiterSurvive? new[] { InfiniteWaiterToolName } : null);
+            
+            fam.FailUnexpectedFileAccesses = false;
+
+            // We instruct the regular test process to spawn InfiniteWaiter as a child
+            var info = ToProcessInfo(
+                ToProcess(
+                    Operation.SpawnExe(
+                        Context.PathTable, 
+                        CreateFileArtifactWithName(InfiniteWaiterToolName, TestDeploymentDir))),
+                fileAccessManifest: fam);
+
+            // Let's shorten the default time to wait for nested processes, since we are spawning 
+            // a process that never ends and we don't want this test to wait for that long
+            info.NestedProcessTerminationTimeout = TimeSpan.FromMilliseconds(10);
+
+            var result = RunProcess(info).GetAwaiter().GetResult();
+            XAssert.AreEqual(0, result.ExitCode);
+
+            if (!letInfiniteWaiterSurvive)
+            {
+                // If we didn't let infinite waiter escape, we should have killed it when the job object was finalized
+                XAssert.IsTrue(result.Killed);
+                result.SurvivingChildProcesses.Any(process => process?.Path != null && string.Equals(System.IO.Path.GetFileName(process.Path), InfiniteWaiterToolName, StringComparison.InvariantCultureIgnoreCase));

```

Comment:
Don't you want to assert what this call to `Any()` should return?

Instead of asserting whether this `Any` returns true or false, you could use something like

```cs
XAssert.Contains(
    result.SurvivingChildProcesses.Select(p => p?.Path).Where(p => p != null).Select(p => Path.GetFileName(p).ToUpperInvariant()),
    InfiniteWaiterToolName.ToUpperInvariant());
```

Reply:
Sure, will change

---
In reply to: [339159918](https://github.com/microsoft/BuildXL/pull/1099#discussion_r339159918) [](ancestors = 339159918)